redisson/src/main/java/org/redisson/RedissonLocalCachedMap.java

```diff
@@ -80,12 +80,12 @@ public RedissonLocalCachedMap(Codec codec, CommandAsyncExecutor connectionManage
 
     private void init(LocalCachedMapOptions<K, V> options, EvictionScheduler evictionScheduler) {
         if (options.getCacheProvider() == LocalCachedMapOptions.CacheProvider.CAFFEINE) {
-            options.storeCacheKey(false);
+            options.useObjectAsCacheKey(false);
```

Comment:
IllegalArgumentException should be thrown if this option can't be used with specific cache provider.

Reply:
Done